src/xtgeo/grid3d/_gridprop_import_grdecl.py

```diff
@@ -89,9 +101,21 @@ def read_grdecl_3d_property(filename, keyword, dimensions, dtype=float):
     return np.ascontiguousarray(f_order_values.reshape(dimensions, order="F"))
 
 
-def import_grdecl_prop(pfile, name, grid):
-    """Read a GRDECL ASCII property record"""
-    result = {}
+def import_grdecl_prop(pfile: _XTGeoFile, name: str, grid: Grid) -> dict[str, Any]:
+    """Import prop for ascii files with GRDECL layout.
+
+    Args:
+        pfile (_XTgeoCFile): xtgeo file instance
+        name (str): Name of parameter.
+        grid (Grid()): XTGeo Grid instance.
+
+    Raises:
+        KeywordNotFoundError: Cannot find property...
+
+    Returns:
+        GridProperty parameter dictionary.
+    """
+    result: dict[str, Any] = {}
```

Comment:
I gave it a shot and it's a bit rough because it spreads into a bunch of other files. I think it might be a good idea to hold off with it for the moment. 

The reason being, this dictionary is returned to be used by the GridProperty constructor which goes through the maze of backward compatibility functions for the deprecated initializer and deprecated `GridProperty().from_file()` method. We just `Any` the kwargs coming out of this dict (and other factory methods that return similar dicts) right now.

I think it will be 100x easier to address when we complete the xtgeo 4.0 deprecations and would naturally fit under that work too.